engine/engine_impl.go

```diff
@@ -75,31 +76,18 @@ func (e *engine) Setup(ctx context.Context, spec *Spec) error {
 		return err
 	}
 
-	logger.FromContext(ctx).
-		WithField("hostname", spec.Server.Name).
-		WithField("user", spec.Server.User).
-		WithField("ip", instance.IP).
-		WithField("id", instance.ID).
-		Debug("dial the server")
-
 	// establish an ssh connection with the server instance
 	// to setup the build environment (upload build scripts, etc)
-	client, err := dial(
+
+	client, err := retryDial(
+		ctx,
 		spec.ip,
 		spec.Server.User,
 		e.privatekey,
 	)
 	if err != nil {
-		logger.FromContext(ctx).
-			WithError(err).
-			WithField("hostname", spec.Server.Name).
-			WithField("user", spec.Server.User).
-			WithField("ip", instance.IP).
-			WithField("id", instance.ID).
-			Debug("failed to dial server")
 		return err
 	}
-	defer client.Close()
```

Comment:
any reason this is being removed?  I believe we still need to close the client connection once we are done.

Reply:
this actually may be safer:

```
	client, err := e.retryDial(ctx, spec)
	if client != nil {
		defer client.Close()
	}
```

we use the above in our macstadium runner

Reply:
This was probably a mistake. I'll update.